roles/development/files/bin/make-changelog.py

```diff
@@ -106,44 +153,78 @@ def main(username, project, version):
         relstr = "Merge branch 'release"
         pullstr = "Merge pull request #"
 
-        commits = [(slug, msg) for slug, msg in commits if relstr not in msg]
-        pulls = [(slug, msg) for slug, msg in commits if pullstr in msg]
-        commits = [(slug, msg) for slug, msg in commits if pullstr not in msg]
+        if pagure:
+            pullstr = "Merge #"
+
+        commits = [
+            (slug, comdate, msg)
+            for slug, comdate, msg in commits
+            if relstr not in msg]
+        pulls = [
+            (slug, msg)
+            for slug, comdate, msg in commits
+            if pullstr in msg]
+        commits = [
+            (slug, comdate, msg)
+            for slug, comdate, msg in commits
+            if pullstr not in msg]
 
         print
         print start
         print "-" * len(start)
 
+        if pagure:
+            pulls.extend(get_pull_request_pagure(project, commits[0]))
+
         if pulls:
             print
             print "Pull Requests"
             print
 
-        for slug, msg in pulls:
-            number = msg[len(pullstr):].split()[0]
-            try:
-                title, author, link = get_pull_info(username, project, number)
+        if pagure:
+            for number, title, author in pulls:
                 author = "(@%s)" % author
-            except KeyError as e:
-                sys.stderr.write('Problems getting info for '
-                                 '#%s\n%r\n' % (number, e))
-                # Some fallbacks
-                author = ''
-                title = msg
-                link = pull_url_for(username, project, number)
+                link = pull_url_for_pagure(project, number)
 
-            print "- %s #%s, %s\n  %s" % (author.ljust(17), number, title, link)
+                print "- %s #%s, %s\n  %s" % (author.ljust(17), number, title, link)
+
+        else:
+            for slug, msg in pulls:
```

Comment:
This will get very confusing very fast.

So, pagure pulls are composed of ``(number, title, author)`` tuples but github pulls are composed of ``(slug, msg)`` tuples?

:-1: to maintaining two different informal schemas here in the script.  They'll need to be made symmetric so the logic can be streamlined here.